inventory.lua

```diff
@@ -1,31 +1,52 @@
-local function sendMessage(pos, msg, channel)
-	if channel == nil then
-		channel = minetest.get_meta(pos):get_string("channel")
+-- Sends a message onto the Digilines network.
+-- pos: the position of the Digilines chest node.
+-- action: the action string indicating what happened.
+-- stack: the ItemStack that the action acted on (optional).
+-- side: which side of the chest the action occurred (optional).
+local function sendMessage(pos, action, stack, side)
+	local channel = minetest.get_meta(pos):get_string("channel")
+	local msg = {action = action}
+	if stack ~= nil then
+		msg.stack = stack:to_table()
+	end
+	if side ~= nil then
+		-- Duplicate the vector in case the caller expects it not to change.
+		msg.side = vector.new(side)
 	end
 	digilines.receptor_send(pos,digilines.rules.default,channel,msg)
 end
 
-local function maybeString(stack)
-	if type(stack)=='string' then return stack
-	elseif type(stack)=='table' then return dump(stack)
-	else return stack:to_string()
+-- Checks if the inventory has become empty and, if so, sends an empty message.
+local function checkEmpty(pos, listname)
+	if minetest.get_meta(pos):get_inventory():is_empty(listname) then
+		sendMessage(pos, "empty")
 	end
 end
 
-local function can_insert(pos, stack)
-	local can = minetest.get_meta(pos):get_inventory():room_for_item("main", stack)
-	if can then
-		sendMessage(pos,"put "..maybeString(stack))
-	else
-		-- overflow and lost means that items are gonna be out as entities :/
-		sendMessage(pos,"lost "..maybeString(stack))
+-- Checks if the inventory has become full for a particular type of item and,
+-- if so, sends a full message.
+local function checkFull(pos, listname, stack)
+	local oneItemStack = ItemStack(stack)
+	oneItemStack:set_count(1)
+	if not minetest.get_meta(pos):get_inventory():room_for_item(listname, oneItemStack) then
+		sendMessage(pos, "full", oneItemStack)
 	end
-	return can
 end
 
 local tubeconn = minetest.get_modpath("pipeworks") and "^pipeworks_tube_connection_wooden.png" or ""
 local tubescan = minetest.get_modpath("pipeworks") and function(pos) pipeworks.scan_for_tube_objects(pos) end or nil
 
+-- A place to remember things from allow_metadata_inventory_put to
+-- on_metadata_inventory_put.
+local lastInventoryPutListname = nil
+local lastInventoryPutIndex = nil
+local lastInventoryPutStack = nil
+
+-- A place to remember things from allow_metadata_inventory_take to
+-- tube.remove_items.
+local lastInventoryTakeListname = nil
+local lastInventoryTakeIndex = nil
```

Comment:
>**Functions and variables should be named in lowercase_underscore_style**

http://dev.minetest.net/Lua_code_style_guidelines#Naming

Reply:
Sorry. I was following the naming convention of the existing `sendMessage` function. Want me to change that one too?

Reply:
Assuming the answer is yes, fixed in `ca54a52b0521eda8f6d76a45a07755f71b0caadc`.